src/Build/Evaluation/Expander.cs

```diff
@@ -4677,6 +4704,40 @@ private static bool TryGetArgs(object[] args, out string arg0, out int arg1)
                 return false;
             }
 
+            private static bool IsFloatingPointRepresentation(object value)
+            {
+                const char numberDecimalSeparator = '.';
+
+                return value is double || (value is string str && str.Contains(numberDecimalSeparator));
```

Comment:
`long.TryParse` would always be attempted first and values that are `double` will always have the overhead of trying to parse twice. I was under the impression that parsing twice for `double` would be an issue from a performance perspective. But I think that was probably a premature optimization. The `double` overloads are used much less than the `long` overloads.

The `IsFloatingPointRepresentation` method is not intended to be 'air tight' and a false negative is not an issue in terms of behavior. But in terms of code maintenance, that's an easily missed nuance.

I agree that it would be better to eliminate the `IsFloatingPointRepresentation` method.

Reply:
Perf and maintainability is a delicate balance. I wish we had a magic formula to decide :-)
There was a consensus on the advice from @ladipro

Reply:
on the core libraries level we continue to make TryParse faster. I do not think it is possible to do better and still have correctness. I think it is usually best to defer to the libraries and open perf bugs against them if necessary 😄

Reply:
new commit to make changes to use TryParse